src/llmcompressor/transformers/finetune/text_generation.py

```diff
@@ -65,27 +69,36 @@ def train(**kwargs):
     """
     CLI entrypoint for running training
     """
-    model_args, data_args, training_args = parse_args(**kwargs)
+    model_args, data_args, recipe_args, training_args, _ = parse_args(
+        include_training_args=True, **kwargs
+    )
     training_args.do_train = True
-    main(model_args, data_args, training_args)
+    main(model_args, data_args, recipe_args, training_args)
 
 
 def eval(**kwargs):
     """
     CLI entrypoint for running evaluation
     """
-    model_args, data_args, training_args = parse_args(**kwargs)
+    model_args, data_args, recipe_args, training_args, _ = parse_args(
+        include_training_args=True, **kwargs
+    )
     training_args.do_eval = True
-    main(model_args, data_args, training_args)
+    main(model_args, data_args, recipe_args, training_args)
 
 
 def oneshot(**kwargs):
     """
     CLI entrypoint for running oneshot calibration
     """
-    model_args, data_args, training_args = parse_args(**kwargs)
+    # TODO: Get rid of training args when Oneshot refactor comes in
+    model_args, data_args, recipe_args, training_args, output_dir = parse_args(
+        include_training_args=True, **kwargs
```

Comment:
If the purpose of separating training args is to throw an error if the user passes training args not relevant to oneshot, then shouldn't we not include training args for oneshot?

Reply:
The change where you set `include_training_args=False` is not in the main refactor PR. I don't know when that change is supposed to be made.

Reply:
You are right that we should make the main function for oneshot not dependent on training_args. Making those changes will really be just for oneshot, which will be refactored in the follow up PR anyways. So if i make changes it will only last until the Oneshot main refactor lands.

Reply:
Why can't that change be logically grouped into this PR. That way you know that the changes in this PR are worth it because someone can use this branch to test that the desired feature actually works.

Reply:
> Making those changes will really be just for oneshot, which will be refactored in the follow up PR anyways. So if i make changes it will only last until the Oneshot main refactor lands.

I thought the overall purpose of these changes was to prepare for the oneshot main PR? If you plan on overwriting the changes anyways, then why are you proposing them?

Reply:
I'll note that there used to be a *purpose* section in the PR description but has since been deleted

Reply:
> Making those changes will really be just for oneshot, which will be refactored in the follow up PR anyways. So if i make changes it will only last until the Oneshot main refactor lands.

I don't understand why these changes are being proposed if you think they're not relevant to the oneshot main refactor

Reply:
Yeah im actually going to review the new logic here